agent/agent.go

```diff
@@ -550,11 +550,14 @@ func (a *Agent) Publisher(ctx context.Context, subscriptionID string) (exec.LogP
 func (a *Agent) nodeDescriptionWithHostname(ctx context.Context, tlsInfo *api.NodeTLSInfo) (*api.NodeDescription, error) {
 	desc, err := a.config.Executor.Describe(ctx)
 
-	// Override hostname and TLS info
+	// Override hostname
 	if desc != nil {
-		if a.config.Hostname != "" && desc != nil {
+		if a.config.Hostname != "" {
 			desc.Hostname = a.config.Hostname
 		}
+		if desc.Resources != nil && a.config.GenericResources != nil {
+			desc.Resources.Generic = a.config.GenericResources
```

Comment:
The current behavior for the existing resources (`NanoCPUs` and `MemoryBytes`) is that they get filled in by `Describe` (see agent/exec/dockerapi/executor.go for the `dockerapi` executor implementation). I think we should follow this with generic resources, so that they are consistent with the existing resources, and because it's an easy way to address the hotplug case (`Describe` is called periodically, and the node is updated dynamically based on the result).